StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1308CodeFixProvider.cs

```diff
@@ -52,9 +52,28 @@ public override FixAllProvider GetFixAllProvider()
                     continue;
                 }
 
+                var numberOfCharsToRemove = 2;
+
+                // If a variable contains multiple prefixes that would result in this diagnostic, 
+                // we should detect that and remove all of the bad prefixes such that after 
+                // the fix is applied there are no more violations of this rule.
+                for (int i = 2; i < token.ValueText.Length; i += 2)
+                {
+                    var nextTwoChars = token.ValueText.Substring(i, 2);
```

Comment:
Consider using [this overload of `string.Compare`](https://msdn.microsoft.com/en-us/library/7aaf32ef.aspx) with `StringComparison.Ordinal` to avoid allocating new strings with `Substring`.